### ISSUE TITLE
Websocket subscriptions

### ISSUE THREAD

**Issue by justinmoon** [fedimint/fedimint#2708]
All federation APIs use websockets, but we don't support websocket subscriptions (the whole reason we chose to use websockets for our APIs!)

**Comment by Maan2003** [comment]
I don't think this is useful for one subscription-one response type of notification. Because that looks same on the network level.

It is useful for one subscription-multiple responses type of notification. I can't think of instances of this. 
If you know of any examples where this will be useful, please tell.

**Comment by elsirion** [comment]
> Because that looks same on the network level.

It might look the same on the wire, but is it handled the same by `jsonrpsee`? One problem we had was running into timeouts for RPC requests. Another was having a client side limit on parallel requests. I wonder if subscriptions would be the better fit just because their anticipated usage pattern matches more closely to how we'd use them.

**Comment by douglaz** [comment]
The decrypted preimage status call is one example prone to subscriptions:
https://github.com/fedimint/fedimint/pull/3939/files

We have `Pending -> Some | Invalid` (plus a potential `Timeout` status).

**Comment by elsirion** [comment]
Our plant to switch to Iroh long terms makes this obsolete imo.
